tests/unit/test_charm.py

```diff
@@ -31,115 +31,88 @@ def setUp(self, network_mock):
         self.addCleanup(self.harness.cleanup)
         self.harness.begin()
 
-    @mock.patch("charms.operator_libs_linux.v1.systemd.service_start")
+    @mock.patch("charms.operator_libs_linux.v2.snap.Snap.start")
     def test_pollen_start(self, start_mock):
         PollenService.start()
-        start_mock.assert_called_with("pollen.service")
+        start_mock.assert_called_once()
 
-    @mock.patch("charms.operator_libs_linux.v1.systemd.service_stop")
-    def test_pollen_stop(self, start_mock):
+    @mock.patch("charms.operator_libs_linux.v2.snap.Snap.stop")
+    def test_pollen_stop(self, stop_mock):
         PollenService.stop()
-        start_mock.assert_called_with("pollen.service")
+        stop_mock.assert_called_once()
 
     @mock.patch("glob.glob")
     @mock.patch(
         "builtins.open",
         mock.mock_open(read_data='RNGDOPTIONS="--fill-watermark=90% --feed-interval=1"'),
     )
     @mock.patch("charms.operator_libs_linux.v1.systemd.service_restart")
-    @mock.patch("charms.operator_libs_linux.v1.systemd.service_reload")
     @mock.patch("subprocess.run")
-    @mock.patch("shutil.chown")
-    @mock.patch("shutil.copy")
-    @mock.patch("os.path.exists")
-    @mock.patch("os.mkdir")
+    @mock.patch("pathlib.Path.exists")
     @mock.patch("charms.operator_libs_linux.v0.apt.add_package")
     @mock.patch("charms.operator_libs_linux.v0.apt.update")
```

Comment:
This looks like it needs to be refactored with all these `mock.patch`, potentially in a follow up story?

Reply:
Sounds good, I'll create a story later for the refactoring to take place.